### PR TITLE
Add WebAssembly SDK recipe and `make-wasm-sdk` subcommand

### PR DESCRIPTION
**Pull request by kateinoigakukun**
This patch introduces an experimental WebAssembly SDK recipe and `make-wasm-sdk` subcommand. The make command takes host Swift toolchain package, target Swift stdlib (`lib/swift{,_static}`), and WASI sysroot like follows:

```console
$ # Install https://github.com/swiftwasm/swift/releases/download/swift-wasm-DEVELOPMENT-SNAPSHOT-2024-01-18-a/swift-wasm-DEVELOPMENT-SNAPSHOT-2024-01-18-a-macos_arm64.pkg

$ swift run swift-sdk-generator make-wasm-sdk \
      --target wasm32-unknown-wasi --sdk-name DEVELOPMENT-SNAPSHOT-2024-01-18-a_wasm \
      --host-swift-package-path /Library/Developer/Toolchains/swift-wasm-DEVELOPMENT-SNAPSHOT-2024-01-18-a.xctoolchain \
      --target-swift-package-path /Library/Developer/Toolchains/swift-wasm-DEVELOPMENT-SNAPSHOT-2024-01-18-a.xctoolchain \
      --wasi-sysroot /Library/Developer/Toolchains/swift-wasm-DEVELOPMENT-SNAPSHOT-2024-01-18-a.xctoolchain/usr/share/wasi-sysroot
```

### CODE REVIEW

File: Sources/GeneratorCLI/GeneratorCLI.swift [hunk 1]
```diff
@@ -222,6 +223,56 @@ extension GeneratorCLI {
...
+
+    @OptionGroup
+    var generatorOptions: GeneratorOptions
+
+    @Option(
+      help: """
+      Path to the Swift toolchain package containing the Swift compiler that runs on the host platform.
+      """
+    )
+    var hostSwiftPackagePath: String
```
Comment (euanh) [inline Sources/GeneratorCLI/GeneratorCLI.swift:244]: Should `--host-swift-package-path` and `--target-swift-package-path` be common options between different SDK generators?   The Linux generator currently assumes `Artifacts` and `Bundles`, but it also downloads the archives and unpacks them.

Comment (kateinoigakukun) [reply Sources/GeneratorCLI/GeneratorCLI.swift:244]: Yes, they could be common options. Let me open a separate PR for it.

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by kateinoigakukun** [comment]
@swift-ci test

**Review by MaxDesiatov — APPROVED** [review]
LGTM! I realize this is hard to test without downloading the actual dependencies on CI. Eventually we should have a way to pass a virtual file system to the generator engine for testing purposes. But for that we need to resolve https://github.com/apple/swift-sdk-generator/issues/76 first.

**Comment by kateinoigakukun** [comment]
I think it would be worth considering standalone stdlib build support in the generator to avoid such external dependencies in CI :)

**Comment by MaxDesiatov** [comment]
> standalone stdlib build support in the generator

Can you elaborate on what it would mean in this context?

**Comment by kateinoigakukun** [comment]
I mean if we can build stdlib on sdk-generator CI, we don't need to download stdlib artifacts from third party swiftwasm releases.

**Comment by kateinoigakukun** [comment]
@swift-ci Please test

**Review by euanh — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Pull request merged**